### PR TITLE
[Refactor] Wait for apply finish when exec primary key publish version

### PR DESCRIPTION
**Pull request by sevev**
Signed-off-by: zhangqiang <qiangzh95@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

PrimaryKey table only submit apply task to thread pool and return success directly when we do publish version right now. If we query immediately after publish success, we may need to wait for version apply finish which cause unstable query time.
So we try to wait for apply finished when we exec publish version.

It is important to note that this solution is a temporary one and does not completely solve the problem of unstable query time, i will reconsider the primary key publish version mechanism and solve the problem in the future pr.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -915,5 +915,6 @@ CONF_String(rocksdb_cf_options_string, "block_based_table_factory={block_cache=1
 
 // limit local exchange buffer's memory size per driver
 CONF_Int64(local_exchange_buffer_mem_limit_per_driver, "134217728"); // 128MB
+CONF_mInt64(wait_apply_time, "6000")                                 // 6s
```
Comment (chaoyli) [inline be/src/common/config.h:918]: It means the query may stuck for six seconds?
If the bug fix is not critical for users, I think we can better to design a better solution.

Comment (sevev) [reply be/src/common/config.h:918]: > It means the query may stuck for six seconds? If the bug fix is not critical for users, I think we can better to design a better solution.

If there are idle threads, publish version thread will wait up to 6s. If all threads are running, we don't wait and return immediately just like before.

File: be/src/storage/tablet_updates.cpp [hunk 2]
```diff
@@ -1097,6 +1100,9 @@ Status TabletUpdates::_wait_for_version(const EditVersion& version, int64_t time
     if (!(_edit_version_infos[_apply_version_idx]->version < version)) {
         return Status::OK();
     }
+    if (timeout_ms == 0) {
```
Comment (luohaha) [inline be/src/storage/tablet_updates.cpp:1103]: When `timeout_ms == 0`, then  `_wait_for_version` just return OK may break the semantics of `_wait_for_version`. If someone who want to check version and not need to wait, he will just get OK by mistake. 

You can do this check outside `_wait_for_version`, eg:
```
if (wait_time > 0) {
   st = _wait_for_version(EditVersion(version, 0), wait_time, ul);
}
```

Comment (sevev) [reply be/src/storage/tablet_updates.cpp]: > When `timeout_ms == 0`, then `_wait_for_version` just return OK may break the semantics of `_wait_for_version`. If someone who want to check version and not need to wait, he will just get OK by mistake.
> 
> You can do this check outside `_wait_for_version`, eg:
> 
> ```
> if (wait_time > 0) {
>    st = _wait_for_version(EditVersion(version, 0), wait_time, ul);
> }
> ```

Great advice, changed in https://github.com/StarRocks/starrocks/pull/17101/commits/408f82035b6a4476d50d0f24dd37fdbeb6a3dcf3

**Review by luohaha — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**